integration/unplugin/src/index.ts

```diff
@@ -238,26 +233,86 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
       if (!isCivetTranspiled(id)) return null;
 
       const filename = path.resolve(process.cwd(), id.slice(0, -outExt.length));
-      const code = await fs.promises.readFile(filename, 'utf-8');
+      const rawCivetSource = await fs.promises.readFile(filename, 'utf-8');
       this.addWatchFile(filename);
 
-      // Ideally this should have been done in a `transform` step
-      // but for some reason, webpack seems to be running them in the order
-      // of `resolveId` -> `loadInclude` -> `transform` -> `load`
-      // so we have to do transformation here instead
-      const compiled = civet.compile(code, {
-        // inlineMap: true,
-        filename: id,
-        js: transpileToJS,
-        sourceMap: true,
-      });
+      let compiled: {
+        code: string;
+        sourceMap: SourceMap | string;
+      } = undefined!;
+
+      if (options.ts === 'civet' && !transformTS) {
+        compiled = civet.compile(rawCivetSource, {
+          filename: id,
+          js: true,
+          sourceMap: true,
+        });
+      } else {
+        const compiledTS = civet.compile(rawCivetSource, {
+          filename: id,
+          js: false,
+          sourceMap: true,
+        });
```

Comment:
We shouldn't do this call when `options.ts === "civet"`: we don't want to run the Civet transpiler twice in this case. (Part of the point is for it to run faster! Also it's a little dicey whether the output of Civet is also valid input to Civet, as we don't necessarily preserve indentation.)

Reply:
Ah, I see. We could modify the compiler to be able to output both versions for this case; this just involves [running generate](https://github.com/DanielXMoore/Civet/blob/0efd56eb5fff8f35173a17623064279753409476/source/main.civet#L149) on the same parse tree with two different options.

(There's also still the double-transpilation issue mentioned below.)